tests/ssg_test_suite/rule.py

```diff
@@ -60,27 +60,44 @@ def get_viable_profiles(selected_profiles, datastream, benchmark):
     return valid_profiles
 
 
-def _send_scripts(rule_dir, domain_ip, *scripts_list):
-    """Upload scripts to VM."""
-    # scripts_list is list of absolute paths
-    remote_dir = './'
+def _send_scripts(domain_ip):
+    remote_dir = './ssgts'
+    archive_file = data.create_tarball('.')
+    remote_archive_file = os.path.join(remote_dir, archive_file)
     machine = "root@{0}".format(domain_ip)
-    logging.debug("Uploading scripts {0}".format(scripts_list))
-    rule_name = os.path.basename(rule_dir)
-    log_file_name = os.path.join(LogHelper.LOG_DIR, rule_name + ".upload.log")
+    logging.debug("Uploading scripts.")
+    log_file_name = os.path.join(LogHelper.LOG_DIR, "data.upload.log")
+
+    command = "ssh {0} mkdir -p {1}".format(machine, remote_dir)
+    with open(log_file_name, 'a') as log_file:
+        try:
+            subprocess.check_call(shlex.split(command),
+                                  stdout=log_file,
+                                  stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            logging.error("Cannot create directoru {0}.".format(remote_dir))
+            return False
 
-    command = "scp {0} {1}:{2}".format(' '.join(scripts_list),
-                                       machine,
-                                       remote_dir)
+    command = "scp {0} {1}:{2}".format(archive_file, machine, remote_dir)
     with open(log_file_name, 'a') as log_file:
         subprocess.check_call(shlex.split(command),
                               stdout=log_file,
                               stderr=subprocess.STDOUT)
 
+    command = "ssh {0} tar xf {1} -C {2}".format(machine, remote_archive_file, remote_dir)
```

Comment:
You can continue using that `log_file` object as long as you want, no need to get it again in lines below.